index.js

```diff
@@ -8,6 +13,81 @@ server.use(helmet());
 
 // endpoints here
 
+// C
+server.post('/api/zoos', async (req, res) => {
+  try {
+    const [ id ] = await db('zoos')
+      .insert(req.body);
+    const zoo = await db('zoos')
+      .where({ id })
+      .first();
+    res.status(201).json(zoo)
+  } catch (err) {
+    res.status(500).json({ message: 'Could not add zoo' })
+  }
+})
+
+// R
+server.get('/api/zoos', async (req, res) => {
+  try {
+    const zoos = await db('zoos');
+    res.status(200).json(zoos)
+  } catch (err) {
+    res.status(500).json({ message: "Could not retrieve zoos" })
+  }
+})
+
+server.get('/api/zoos/:id', async (req, res) => {
+  try {
+    const zoo = await db('zoos')
+    .where({ id: req.params.id})
+    if (zoo.length === 0) {
+      res.status(404).json({ error: "That id does not exist" })
+    } else {
+      res.status(200).json(zoo);
+    }
+  } catch (err) {
+    res.status(500).json({ message: "Could not retrieve requested zoo" })
+  }
+})
+
+// U
+server.put('/api/zoos/:id', async (req, res) => {
+  try {
+    const rec = await db('zoos')
+    .where({ id: req.params.id })
+    .update(req.body);
+    if (rec === 0) {
+      res.status(404).json({ error: "That id does not exist" })
+    } else {
+      const zoo = await db('zoos')
+      .where({ id: req.params.id })
+      .first();
+      res.status(200).json(zoo);
+    }
+  } catch (err) {
+    res.status(500).json(err);
+  }
+})
+
+// D
+server.delete('/api/zoos/:id', async (req, res) => {
+  try {
+    const recs = await db('zoos')
+    .where({ id: req.params.id })
+    .del()
+    if (recs === 0) {
+      res.status(404).json({ error: "That id does not exist" })
+    } else {
+      res.status(204).end();
+    }
+  } catch (err) {
+    res.status(500).json({ message: "Could not complete delete request" })
+  }
+})
+
+
+
```

Comment:
nice work on leaving comments! I would use at least a full word though instead of a single letter.